tests/mpi/periodicity_08.cc

```diff
@@ -0,0 +1,184 @@
+
+  // rescale to [0,1]^2 for this to work:
+  for (auto &p : points)
+    {
+      p[0] /= 1e6;
+      p[1] /= 5e5;
+    }
+
+  for (auto &cell : triangulation.active_cell_iterators())
+    {
+      cell->set_coarsen_flag();
+      for (auto &p : points)
+        if (cell->point_inside(p))
+          {
+            cell->clear_coarsen_flag();
+            cell->set_refine_flag();
+          }
+    }
+  triangulation.execute_coarsening_and_refinement();
+
+  pcout << "number of elements: " << triangulation.n_global_active_cells()
+        << std::endl;
+
+  // create dof_handler
+  FESystem<dim>   FE(FE_Q<dim>(QGaussLobatto<1>(2)), 1);
+  DoFHandler<dim> dof_handler(triangulation);
+  dof_handler.distribute_dofs(FE);
+
+  // write mesh for visualization
+  DataOut<dim> data_out;
+  data_out.attach_dof_handler(dof_handler);
+  Vector<float> subdomain(triangulation.n_active_cells());
+  for (unsigned int i = 0; i < subdomain.size(); ++i)
+    subdomain(i) = triangulation.locally_owned_subdomain();
+  data_out.add_data_vector(subdomain, "subdomain");
+  data_out.build_patches();
+  data_out.write_vtu_in_parallel(std::string("mesh.vtu").c_str(),
+                                 mpi_communicator);
+
+  IndexSet locally_relevant_dofs;
+  DoFTools::extract_locally_relevant_dofs(dof_handler, locally_relevant_dofs);
+
+  IndexSet locally_active_dofs;
+  DoFTools::extract_locally_active_dofs(dof_handler, locally_active_dofs);
+
+  const std::vector<IndexSet> locally_owned_dofs =
+    Utilities::MPI::all_gather(MPI_COMM_WORLD,
+                               dof_handler.locally_owned_dofs());
+
+  std::map<types::global_dof_index, Point<dim>> supportPoints;
+  DoFTools::map_dofs_to_support_points(MappingQ1<dim>(),
+                                       dof_handler,
+                                       supportPoints);
+
+  /// creating combined hanging node and periodic constraint matrix
+  AffineConstraints<double> constraints;
+  constraints.clear();
+  constraints.reinit(locally_relevant_dofs);
+
+  for (int d = 0; d < dim; ++d)
+    DoFTools::make_periodicity_constraints(
+      dof_handler, 2 * d, 2 * d + 1, d, constraints);
+
+  const bool consistent =
+    constraints.is_consistent_in_parallel(locally_owned_dofs,
+                                          locally_active_dofs,
+                                          mpi_communicator,
+                                          /*verbose*/ true);
+
+  pcout << "Periodicity constraints are consistent in parallel: " << consistent
+        << std::endl;
+
+  DoFTools::make_hanging_node_constraints(dof_handler, constraints);
+
+  const bool hanging_consistent =
+    constraints.is_consistent_in_parallel(locally_owned_dofs,
+                                          locally_active_dofs,
+                                          mpi_communicator);
+
+  pcout << "Hanging nodes constraints are consistent in parallel: "
+        << hanging_consistent << std::endl;
```

Comment:
Why aren't you writing to deallog?

Reply:
because this was a copy&paste from an older test. This is now fixed.